pkg/driver/node.go

```diff
@@ -118,40 +119,67 @@ func (d *nodeService) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 	}
 
 	klog.V(5).Infof("NodePublishVolume: mounting juicefs with secret %+v, options %v", reflect.ValueOf(secrets).MapKeys(), mountOptions)
-	jfs, err := d.juicefs.JfsMount(secrets, mountOptions)
+	jfs, err := d.juicefs.JfsMount(volumeID, secrets, mountOptions)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "Could not mount juicefs: %v", err)
 	}
 
-	bindSource := jfs.GetBasePath()
-	if subPath, ok := req.GetVolumeContext()["subPath"]; ok {
-		bindSource = path.Join(bindSource, subPath)
+	var (
+		subPath string
+		volRoot string
+	)
+	if m, ok := volCtx["mode"]; ok && m == "dynamic" {
+		// dynamic provisioning
+		// /jfs/$(subPath)/$(volumeID) if subPath
```

Comment:
This is inconsistent with

```
	// subPath is not supported in dynamic provisioning
 	delete(volCtx, "subPath")
```

Reply:
If `subPath` is not supported by dynamic provisioning, we can simply assign `volumeID` to `subPath` to make the logic consistent. If `subPath` is supported, concat `subPath` and `volumeID` to make the directory for dynamic provisioning, and `subPath` only when static. This would allow you to substitute `CreateVol` with `Mkdir`.

What do you think?

Reply:
+1

Reply:
i was just leaving code here in case we'll support `subPath` in the future (by creating volume in controller driver)...

Reply:
since `subPath` is not supported in dynamic provisioning, i removed `volRoot` parameter in `CreateVol`, now `subPath` means the relative path from JFS root.